### PR TITLE
Avoid possible OOM

### PR DESCRIPTION
**Pull request by wzshiming**
Signed-off-by: Shiming Zhang <wzshiming@foxmail.com>

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/github.com/kubeedge/viaduct/pkg/packer/reader.go [hunk 1]
```diff
@@ -27,25 +31,43 @@ func (r *Reader) Read() ([]byte, error) {
...
-			klog.Error("failed to read package header from buffer")
+			klog.Error("failed to read package header from buffer", err)
 		}
 		return nil, err
 	}
+	if HeaderSize != n {
+		err := fmt.Errorf("failed to read package header: invalid header, length want %d, but got %d", HeaderSize, n)
+		klog.Error(err)
+		return nil, err
+	}
```
Comment (gy95) [inline staging/src/github.com/kubeedge/viaduct/pkg/packer/reader.go:45]: `io.ReadFull` will return an error if fewer bytes were read, so i dont think this `if` statement is neccessary when we have already decided whether `err` is nil or not.

File: staging/src/github.com/kubeedge/viaduct/pkg/packer/reader.go [hunk 2]
```diff
@@ -27,25 +31,43 @@ func (r *Reader) Read() ([]byte, error) {
 	}
 
 	headerBuffer := make([]byte, HeaderSize)
-	_, err := io.ReadFull(r.reader, headerBuffer)
+	n, err := io.ReadFull(r.reader, headerBuffer)
 	if err != nil {
 		if err != io.EOF {
-			klog.Error("failed to read package header from buffer")
+			klog.Error("failed to read package header from buffer", err)
 		}
```
Comment (gy95) [inline staging/src/github.com/kubeedge/viaduct/pkg/packer/reader.go:38]: need `%v` here

**Comment by wzshiming** [comment]
@gy95  Thanks, updated

File: staging/src/github.com/kubeedge/viaduct/pkg/packer/reader.go [hunk 3]
```diff
@@ -7,6 +7,10 @@ import (
 	"k8s.io/klog/v2"
 )
 
+const (
+	maxPayloadLen = 32 * 1 << 20 // 32Mi
```
Comment (fisherxu) [inline staging/src/github.com/kubeedge/viaduct/pkg/packer/reader.go:11]: What is the basis of this value？

Comment (wzshiming) [reply staging/src/github.com/kubeedge/viaduct/pkg/packer/reader.go:11]: This type of limit in k8s is generally 10MB, but I feel that this scenario may have a configuration file exceeding 10MB, so increase it to 32MB

**Pull request closed without merging**